### PR TITLE
Added Typescripts types

### LINKED ISSUE #9: TypeScript
I am using this library with TypeScript. Because there are no types, I have written my own. The API is small, so it was not big deal.

I think there may be other people using this library in a TypeScript code base, so I am thinking whether you would accept a **pull request for a TypeScript definition file**?

I am happy to maintain it for future changes. I can also create tests for the types using [tsd](https://www.npmjs.com/package/tsd)

### PR DESCRIPTION
**Pull request by mfpopa**
Fixes #9 

I did not update the package version, but updating just the minor version to `2.1.0` is sufficient because adding types will not break any functionality. Anyone using this package in a TypeScript project would have set their own types anyways, so on `npm i` they'll keep using their own types even though they'll also pull the ones I have added.

Sorry this took so long. I completely forgot about it. Please let me know whenever the API changes and I'll update the types.

Any changes/comments, let me know.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
   "main": "index.js",
```
Comment (bausano) [inline package.json:5]: I've seen projects that define `"types": "index.d.js" property in `package.json`. It's also used in the [Typescript docs for publishing](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html). The compiler probably defaults to the `.d.ts` file with the same name, or to `index.d.ts` though.

**Comment by bausano** [comment]
@131 thoughts on this PR? It'd be amazing to publish the types.

File: package.json (hunk 1 above)
Comment (mfpopa) [reply package.json:5]: I have added the `types` property. In this case it is not necessary, but seems like TypeScript documentation advises to add it: https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#including-declarations-in-your-npm-package

**Comment by bausano** [comment]
@131 Sorry to bug you, but what do you think about this PR? It'd be amazing if it could be published!

**Comment by bartvanremortele** [comment]
@131 would be great if we could merge and publish this

**Comment by hugoduraes** [comment]
@131 Can this be merged?

**Comment by aitorres** [comment]
Bump to see if this can be merged @131

**Comment by oleg-koval** [comment]
Hi @131 , would be great if you could merge & release this one please :)

**Comment by tpimh** [comment]
Does this PR need an update to support `allowDotlessTLD`?

**Comment by 131** [comment]
Looking into this

**Comment by 131** [comment]
Yep, it's missing the allowDotlessTLD typing. Can someone help me here, as i do not speak typescript...
I'll merge that once is fixed

**Pull request closed without merging**